scripts/dts/edtlib.py

```diff
@@ -698,7 +698,7 @@ def bus(self):
     def on_bus(self):
         "See the class docstring"
         bus_node = self.bus_node
-        return bus_node.bus if bus_node else None
+        return bus_node.bus if bus_node else bus_node
```

Comment:
This was due to an unfortunate issue with pylint. For some reason (probably due to my lack of understanding), pylint doesn't understand the type of bus_node.bus (which is to be either NoneType, str, or list). But because of the None in there, it only knows that it can be NoneType. This causing it to trip up with it sees the iteration I do after verifying that it is a list giving the error that it is `E1133: Non-iterable value on_bus is used in an iterating context (not-an-iterable)`. Wasn't quite sure how else to work around this...

Reply:
Sorry for the delay. I needed some time to think about this and I also had a vacation.

pylint's error is confusing but there is a real problem here. People expect this to be a string or None. If you grep around the codebase you will see that making bus return a list or a string or None will break a lot of code, including gen_defines.py:

https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/dts/gen_defines.py#L326

and gen_devicetree_rest.py:

https://github.com/zephyrproject-rtos/zephyr/blob/master/doc/scripts/gen_devicetree_rest.py#L368

TL;DR this approach isn't workable in my opinion. Have you tested it against the internal I3C drivers you're working on? It seems like gen_defines.py in particular should crash.

I think we need a more significant API change than this, something like changing `bus` to `buses` and `on_bus` to `on_buses` in both Node and Binding. Both of these should return lists of strings, which may be empty. Then we need to update the rest of the tree to handle this.

I'll see if I can cook up a draft as part of the general ongoing conversion of this codebase to a third party library.